File: ios/NitroVideoViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(NitroVideoViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(playerId, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(nativeControls, BOOL)
RCT_EXPORT_VIEW_PROPERTY(contentFit, NSString)
RCT_EXPORT_VIEW_PROPERTY(allowsPictureInPicture, BOOL)
RCT_EXPORT_VIEW_PROPERTY(startsPictureInPictureAutomatically, BOOL)
RCT_EXPORT_VIEW_PROPERTY(requiresLinearPlayback, BOOL)
RCT_EXPORT_VIEW_PROPERTY(useExoShutter, BOOL)
RCT_EXPORT_VIEW_PROPERTY(controllerAutoShow, BOOL)

RCT_EXPORT_VIEW_PROPERTY(onPictureInPictureStart, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onPictureInPictureStop, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onFullscreenEnter, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onFullscreenExit, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onFirstFrameRender, RCTDirectEventBlock)

RCT_EXTERN_METHOD(enterFullscreen:(nonnull NSNumber *)reactTag)
RCT_EXTERN_METHOD(exitFullscreen:(nonnull NSNumber *)reactTag)
RCT_EXTERN_METHOD(startPictureInPicture:(nonnull NSNumber *)reactTag)
RCT_EXTERN_METHOD(stopPictureInPicture:(nonnull NSNumber *)reactTag)

@end

File: ios/NitroVideoViewManager.swift
import Foundation
import React
import UIKit

// ─────────────────────────────────────────────────────────────────────────────
// NitroVideoViewManager — RCTViewManager for NitroVideoView in React Native CLI
// ─────────────────────────────────────────────────────────────────────────────

@objc(NitroVideoViewManager)
public final class NitroVideoViewManager: RCTViewManager {

  public override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  public override func view() -> UIView! {
    return NitroVideoView()
  }

  @objc func enterFullscreen(_ reactTag: NSNumber) {
    bridge.uiManager.addUIBlock { _, viewRegistry in
      guard let view = viewRegistry?[reactTag] as? NitroVideoView else { return }
      view.enterFullscreen()
    }
  }

  @objc func exitFullscreen(_ reactTag: NSNumber) {
    bridge.uiManager.addUIBlock { _, viewRegistry in
      guard let view = viewRegistry?[reactTag] as? NitroVideoView else { return }
      view.exitFullscreen()
    }
  }

  @objc func startPictureInPicture(_ reactTag: NSNumber) {
    bridge.uiManager.addUIBlock { _, viewRegistry in
      guard let view = viewRegistry?[reactTag] as? NitroVideoView else { return }
      try? view.startPictureInPicture()
    }
  }

  @objc func stopPictureInPicture(_ reactTag: NSNumber) {
    bridge.uiManager.addUIBlock { _, viewRegistry in
      guard let view = viewRegistry?[reactTag] as? NitroVideoView else { return }
      view.stopPictureInPicture()
    }
  }
}
